iOS-Weather-App/AppDelegate.swift

```diff
@@ -6,31 +6,83 @@
 //
 
 import UIKit
+import CoreLocation
 
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
 
-
+    var window: UIWindow?
+    private var navigationController: UINavigationController?
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        // Override point for customization after application launch.
+
+        window = UIWindow(frame: UIScreen.main.bounds)
+        window?.makeKeyAndVisible()
+
+        navigationController = UINavigationController(rootViewController: makeWeatherController())
+        
+        window?.rootViewController = navigationController
+        
         return true
     }
+}
+
+private extension AppDelegate {
+    func makeWeatherController() -> UIViewController {
+        let weatherController = WeatherViewController()
+        let weatherViewModel = WeatherViewModel()
 
-    // MARK: UISceneSession Lifecycle
+        pushCurrentDetailsController(viewModel: weatherViewModel)
```

Comment:
not a good idea to create a function for 1 clouser

Reply:
I created here two seperate func for clousers because of Clean Code logic, if I remember correctly you mention something if it has like two seperate clousers inside one function you can seperate that function in two but maybe I misunderstood.

Reply:
I created here two seperate func for clousers because of Clean Code logic, if I remember correctly you mention something if it has like two seperate clousers inside one function you can seperate that function in two but maybe I misunderstood.

Reply:
no, I mean, that if you send more than 2 arguments to the functions, you can create a model, which will help you to send a lot of parameters.

as about clousers, if you use more than 2 clousers, you can create struct Handlers {
which contain all you clousers
}
and use next logic
let handlers = Handlers(...)
viewModel.handlers = handlers

but don't create own funciton for each clouser